src/bigint.rs

```diff
@@ -140,18 +188,25 @@ impl PartialOrd for BigUint {
     }
 }
 
+fn cmp_slice(a: &[BigDigit], b: &[BigDigit]) -> Ordering {
+    debug_assert!(a.last() != Some(&0));
+    debug_assert!(b.last() != Some(&0));
+
+    let (s_len, o_len) = (a.len(), b.len());
+    if s_len < o_len { return Less; }
+    if s_len > o_len { return Greater;  }
+
+    for (&self_i, &other_i) in a.iter().rev().zip(b.iter().rev()) {
+        if self_i < other_i { return Less; }
+        if self_i > other_i { return Greater; }
+    }
```

Comment:
On Mon, Nov 30, 2015 at 03:39:22PM -0800, Josh Stone wrote:
> > @@ -140,18 +188,25 @@ impl PartialOrd for BigUint {
> >      }
> >  }
> >  
> > +fn cmp_slice(a: &[BigDigit], b: &[BigDigit]) -> Ordering {
> > +    debug_assert!(a.last() != Some(&0));
> > +    debug_assert!(b.last() != Some(&0));
> > +
> > +    let (s_len, o_len) = (a.len(), b.len());
> > +    if s_len < o_len { return Less; }
> > +    if s_len > o_len { return Greater;  }
> > +
> > +    for (&self_i, &other_i) in a.iter().rev().zip(b.iter().rev()) {
> > +        if self_i < other_i { return Less; }
> > +        if self_i > other_i { return Greater; }
> > +    }
> 
> Let's give these a/b-names instead of s/o/self/other from the original code.

Whoops, missed that - fixed